src/main/java/seedu/address/model/animal/NameContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.model.animal;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Animal}'s {@code Name} matches any of the keywords given.
+ */
+public class NameContainsKeywordsPredicate implements Predicate<Animal> {
+    private final List<String> keywords;
+
+    public NameContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Animal animal) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(animal.getName().fullName, keyword));
```

Comment:
Do add JavaDocs describing the behaviour of this method, specifically when using `Stream::anyMatch`. If I recall correctly, it returns any words that contain a subset of the specified string.